frontend/public/module/k8s/resource.js

```diff
@@ -186,3 +188,48 @@ export const k8sWatch = (kind, query = {}, wsOptions = {}) => {
   wsOptions.path = path;
   return new WSFactory(path, wsOptions);
 };
+
+/**
+ * Use k8sWatch to wait for a resource to get into an expected condition.
+ * Watches for resource by kind, namespace and optional name.
+ * Promise resolves to a new resource version or rejects with a timeout.
+ * @param {K8sKind} kind
+ * @param {K8sResourceKind} resource
+ * @param {(kind: K8sResourceKind) => boolean} checkCondition
+ * @param {number} timeoutInMs
+ * @returns {Promise<K8sResourceKind>}
+ */
+export const k8sWaitForUpdate = (kind, resource, checkCondition, timeoutInMs) => {
+  const { namespace, name, resourceVersion } = resource.metadata;
+
+  if (checkCondition(resource)) {
+    return Promise.resolve(resource);
+  }
+
+  const watcher = k8sWatch(kind, {
+    ns: namespace,
+    resourceVersion,
+  });
+  const closeConnection = () => watcher.destroy(0);
```

Comment:
```suggestion
  const closeConnection = () => watcher.destroy();
```

Reply:
Ah, my TS brain though that it requires a parameter here.. But okay it's a JS optional here.